File: in_Java/0973 K Closest Points to Origin.java
class Solution {
    public int[][] kClosest(int[][] points, int K) {
        Map<Integer, Integer> hm = new HashMap<>();
        for(int i = 0; i < points.length; i++){
            hm.put(i, points[i][0] * points[i][0] + points[i][1] * points[i][1]);
        }
        PriorityQueue<Map.Entry<Integer,Integer>> pq = new PriorityQueue<>((i,j) -> i.getValue() - j.getValue());
        pq.addAll(hm.entrySet());
        int[][] res = new int[K][2];
        int id = 0;
        while(K-- != 0){
            int cur = pq.poll().getKey();
            res[id][0] = points[cur][0];
            res[id++][1] = points[cur][1];
        }
        return res;
    }
}

File: in_Python/0973 K Closest Points to Origin.py
class Solution(object):
    def kClosest(self, points, K):
        """
        :type points: List[List[int]]
        :type K: int
        :rtype: List[List[int]]
        """
        q = []
        qq = []
        for p in points:
            d = p[0] * p[0] + p[1] * p[1]
            heappush(q,(d,p))

        while len(qq) < K:
            heappush(qq,heappop(q))
        res = []
        while K != 0:
            res.append(heappop(qq)[1])
            K -= 1
        return res